framework/auth/core.py

```diff
@@ -481,14 +502,19 @@ def create(cls, username, password, fullname):
         return user
 
     @classmethod
-    def create_unconfirmed(cls, username, password, fullname, do_confirm=True,
+    def create_unconfirmed(cls, username, password, fullname, oauth=None, do_confirm=True,
                            campaign=None):
         """Create a new user who has begun registration but needs to verify
         their primary email address (username).
         """
         user = cls.create(username, password, fullname)
         user.add_unconfirmed_email(username)
         user.is_registered = False
+        if oauth:
+            if user.oauth:
```

Comment:
You shouldn't need this conditional -- `User.oauth` is a `DictionaryField`. It will default to `{}`, so you can call `.update` on it before actually setting it.